staging/src/k8s.io/controller-manager/pkg/clientbuilder/client_builder.go

```diff
@@ -28,11 +30,11 @@ import (
 // TODO: Extract this into a separate controller utilities repo (issues/68947)
 type ControllerClientBuilder interface {
 	Config(name string) (*restclient.Config, error)
-	ConfigOrDie(name string) *restclient.Config
+	ConfigOrDie(ctx context.Context, name string) *restclient.Config
```

Comment:
I think passing a `logger klog.Logger` is the better choice here because it seems unlikely that these functions will ever need a contxt.